File: src3/main/java/com/regenerationforrged/data/worldgen/preset/settings/FeatureSettings.java
package com.regenerationforrged.data.worldgen.preset.settings;

import com.mojang.serialization.Codec;
import com.mojang.serialization.codecs.RecordCodecBuilder;

public class FeatureSettings {

    public static final Codec<FeatureSettings> CODEC = RecordCodecBuilder.create(i -> i.group(
            Codec.BOOL.fieldOf("trees").forGetter(f -> f.trees),
            Codec.BOOL.fieldOf("lakes").forGetter(f -> f.lakes),
            Codec.BOOL.fieldOf("caves").forGetter(f -> f.caves),
            Codec.BOOL.fieldOf("ores").forGetter(f -> f.ores)
    ).apply(i, FeatureSettings::new));

    public final boolean trees;
    public final boolean lakes;
    public final boolean caves;
    public final boolean ores;

    public FeatureSettings(boolean t, boolean l, boolean c, boolean o) {
        this.trees = t;
        this.lakes = l;
        this.caves = c;
        this.ores = o;
    }
}

File: src2/main/kotlin/com/mbz/data/worldgen/preset/settings/FeatureSettings.kt
package com.mbz.data.worldgen.preset.settings

import com.mojang.serialization.Codec
import com.mojang.serialization.codecs.RecordCodecBuilder

data class FeatureSettings(
    var trees: Boolean = true,
    var lakes: Boolean = true,
    var caves: Boolean = true,
    var ores: Boolean = true
) {
    companion object {
        @JvmField
        val CODEC: Codec<FeatureSettings> = RecordCodecBuilder.create { i ->
            i.group(
                Codec.BOOL.fieldOf("trees").forGetter { it.trees },
                Codec.BOOL.fieldOf("lakes").forGetter { it.lakes },
                Codec.BOOL.fieldOf("caves").forGetter { it.caves },
                Codec.BOOL.fieldOf("ores").forGetter { it.ores }
            ).apply(i, ::FeatureSettings)
        }
    }
}
